pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -125,13 +125,56 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		}
 	}
 
-	if infra.Networks.CloudNAT != nil && infra.Networks.CloudNAT.NatIPNames != nil && len(infra.Networks.CloudNAT.NatIPNames) == 0 {
-		allErrs = append(allErrs, field.Invalid(networksPath.Child("cloudNAT", "natIPNames"), infra.Networks.CloudNAT.NatIPNames, "nat IP names cannot be empty"))
+	if infra.Networks.CloudNAT != nil {
+		allErrs = append(allErrs, ValidateCloudNatConfig(infra.Networks.CloudNAT, networksPath)...)
 	}
 
 	return allErrs
 }
 
+// ValidateCloudNatConfig validates the config of the CloudNat. We intentionally keep the validation light, only
+// checking for gotchas (e.g. the port counts having to be powers of two) and obvious errors.
+func ValidateCloudNatConfig(config *apisgcp.CloudNAT, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	cloudNatPath := fldPath.Child("cloudNAT")
+
+	if config != nil && config.NatIPNames != nil && len(config.NatIPNames) == 0 {
+		allErrs = append(allErrs, field.Invalid(cloudNatPath.Child("natIPNames"), config.NatIPNames, "nat IP names cannot be empty."))
+	}
+
+	if config.EndpointIndependentMapping != nil && config.EnableDynamicPortAllocation && config.EndpointIndependentMapping.Enabled {
+		// There is no more fitting field.Error (e.g. field.MutuallyExclusive) so we put the blame on one flag and use the error msg
+		allErrs = append(allErrs, field.Invalid(cloudNatPath.Child("enableDynamicPortAllocation"), config.EnableDynamicPortAllocation, "dynamic port allocation may not be enabled at the same time as endpoint independent mapping."))
```

Comment:
Isn't it just a tad simpler to include this validation inside the next section of EnableDynamicPortAllocation (l. 150)?

--> 
```
if config.EndpointIndependentMapping != nil && config.EndpointIndependentMapping.Enabled
```

I would also blame EIP maybe instead of DPA if I went that route